libp2p/protocols/pubsub/pubsubpeer.nim

```diff
@@ -281,6 +274,13 @@ proc clearSendPriorityQueue(p: PubSubPeer) =
       value = p.rpcmessagequeue.sendPriorityQueue.len.int64,
       labelValues = [$p.peerId])
 
+proc clearNonPriorityQueue*(p: PubSubPeer) =
+  if len(p.rpcmessagequeue.nonPriorityQueue) > 0:
+    p.rpcmessagequeue.nonPriorityQueue.clear()
+
+    when defined(pubsubpeer_queue_metrics):
+      libp2p_gossipsub_non_priority_queue_size.set(labelValues = [$p.peerId], value = 0)
+
```

Comment:
How asynchronous is this operation? That is: How feasible it is that, between the `clear` call and the metrics update, some value is added to the `nonPriorityQueue`? 🤔

Reply:
This proc isn't async and won't be suspended.

Reply:
Oh, that's right. nvm then